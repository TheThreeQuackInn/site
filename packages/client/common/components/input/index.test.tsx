import React from 'react';
import {create} from 'react-test-renderer';
import Input from './';

describe('Input', () => {
    it('renders correctly', () => {
        const onChange = jest.fn();
        const component = create(<Input name="test" label="Test" onChange={onChange} type="text" value="testing" />);
        expect(component).toMatchSnapshot();
    });

    it('renders min/max correctly', () => {
        const onChange = jest.fn();
        const component = create(
            <Input name="test" label="Test" onChange={onChange} type="text" value="testing" min={0} max={8} />,
        );
        expect(component).toMatchSnapshot();
    });

    it('renders placeholder correctly', () => {
        const onChange = jest.fn();
        const component = create(
            <Input
                name="test"
                label="Test"
                onChange={onChange}
                type="text"
                value="testing"
                placeholder="Type here..."
            />,
        );
        expect(component).toMatchSnapshot();
    });
});
