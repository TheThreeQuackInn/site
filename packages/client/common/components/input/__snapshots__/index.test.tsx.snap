// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<div
  className="w-full"
>
  <label
    className="block uppercase tracking-wide text-gray-700 dark:text-white text-xs font-bold mb-1"
    htmlFor="test"
  >
    Test
  </label>
  <input
    className="block w-full bg-gray-200 dark:bg-gray-500 dark:text-white text-gray-700 rounded p-2 md:px-3 md:py-2 mb-3 focus:outline-none focus:shadow-outline focus:bg-white dark:focus:bg-gray-500"
    id="test"
    name="test"
    onChange={[MockFunction]}
    type="text"
    value="testing"
  />
</div>
`;

exports[`Input renders min/max correctly 1`] = `
<div
  className="w-full"
>
  <label
    className="block uppercase tracking-wide text-gray-700 dark:text-white text-xs font-bold mb-1"
    htmlFor="test"
  >
    Test
  </label>
  <input
    className="block w-full bg-gray-200 dark:bg-gray-500 dark:text-white text-gray-700 rounded p-2 md:px-3 md:py-2 mb-3 focus:outline-none focus:shadow-outline focus:bg-white dark:focus:bg-gray-500"
    id="test"
    max={8}
    min={0}
    name="test"
    onChange={[MockFunction]}
    type="text"
    value="testing"
  />
</div>
`;

exports[`Input renders placeholder correctly 1`] = `
<div
  className="w-full"
>
  <label
    className="block uppercase tracking-wide text-gray-700 dark:text-white text-xs font-bold mb-1"
    htmlFor="test"
  >
    Test
  </label>
  <input
    className="block w-full bg-gray-200 dark:bg-gray-500 dark:text-white text-gray-700 rounded p-2 md:px-3 md:py-2 mb-3 focus:outline-none focus:shadow-outline focus:bg-white dark:focus:bg-gray-500"
    id="test"
    name="test"
    onChange={[MockFunction]}
    placeholder="Type here..."
    type="text"
    value="testing"
  />
</div>
`;
